agent/taskresource/credentialspec/credentialspec_windows.go

```diff
@@ -0,0 +1,600 @@
+// removes them from the task
+func (cs *CredentialSpecResource) clearCredentialSpec() {
+	cs.lock.Lock()
+	defer cs.lock.Unlock()
+
+	for key := range cs.credSpecMap {
+		// TODO: Cleanup file on container instance (optional)
+		delete(cs.credSpecMap, key)
+	}
+}
+
+// CredentialSpecResourceJSON is the json representation of the credentialspec resource
+type CredentialSpecResourceJSON struct {
+	TaskARN                 string                               `json:"taskARN"`
+	CreatedAt               *time.Time                           `json:"createdAt,omitempty"`
+	DesiredStatus           *CredentialSpecStatus                `json:"desiredStatus"`
+	KnownStatus             *CredentialSpecStatus                `json:"knownStatus"`
+	RequiredCredentialSpecs map[string][]*apicontainer.Container `json:"credentialSpecResources"`
+	CredSpecMap             map[string]string                    `json:"credSpecMap"`
+	ExecutionCredentialsID  string                               `json:"executionCredentialsID"`
+}
+
+// MarshalJSON serialises the CredentialSpecResourceJSON struct to JSON
+func (cs *CredentialSpecResource) MarshalJSON() ([]byte, error) {
+	if cs == nil {
+		return nil, errors.New("credential specresource is nil")
+	}
+	createdAt := cs.GetCreatedAt()
+	return json.Marshal(CredentialSpecResourceJSON{
+		TaskARN:   cs.taskARN,
+		CreatedAt: &createdAt,
+		DesiredStatus: func() *CredentialSpecStatus {
+			desiredState := cs.GetDesiredStatus()
+			s := CredentialSpecStatus(desiredState)
+			return &s
+		}(),
+		KnownStatus: func() *CredentialSpecStatus {
+			knownState := cs.GetKnownStatus()
+			s := CredentialSpecStatus(knownState)
+			return &s
+		}(),
+		RequiredCredentialSpecs: cs.getRequiredCredentialSpecs(),
+		CredSpecMap:             cs.getCredSpecMap(),
+		ExecutionCredentialsID:  cs.getExecutionCredentialsID(),
+	})
+}
+
+// UnmarshalJSON deserialises the raw JSON to a CredentialSpecResourceJSON struct
+func (cs *CredentialSpecResource) UnmarshalJSON(b []byte) error {
+	temp := CredentialSpecResourceJSON{}
+
+	if err := json.Unmarshal(b, &temp); err != nil {
+		return err
+	}
+
+	if temp.DesiredStatus != nil {
+		cs.SetDesiredStatus(resourcestatus.ResourceStatus(*temp.DesiredStatus))
+	}
+	if temp.KnownStatus != nil {
+		cs.SetKnownStatus(resourcestatus.ResourceStatus(*temp.KnownStatus))
+	}
+	if temp.CreatedAt != nil && !temp.CreatedAt.IsZero() {
+		cs.SetCreatedAt(*temp.CreatedAt)
+	}
+	if temp.RequiredCredentialSpecs != nil {
+		cs.requiredCredentialSpecs = temp.RequiredCredentialSpecs
+	}
+	cs.taskARN = temp.TaskARN
+	cs.executionCredentialsID = temp.ExecutionCredentialsID
+
+	return nil
+}
+
+func (cs *CredentialSpecResource) setCredentialSpecResourceLocation() error {
+	// TODO: Use registry
+	// This should always be available on Windows instances
+	appDataDir := os.Getenv("APPDATA")
+	if appDataDir != "" {
+		cs.credentialSpecResourceLocation = appDataDir
+	} else {
+		tempDir := os.Getenv("TEMP")
```

Comment:
These seems to be a very generic environment name, it might be easily conflicting with some other user defined environment variables. Do you want to revisit this? 

Also, I'm not sure if it is better to define this [here](https://github.com/aws/amazon-ecs-agent/blob/master/agent/config/types.go#L31) also and pass it down here also, so that agent will load all of the required information when it starts up. @sharanyad might have more context here.

Reply:
A different approach has been taken to resolve this issue. Please take a look at the recent changes.